unityfiles/Assets/Scripts/Game/Player/CharacterController2D.cs

```diff
@@ -1,136 +1,172 @@
+		{
 			// If the character has a ceiling preventing them from standing up, keep them crouching
-			if (Physics2D.OverlapCircle(ceilingCheck.position, CEILING_RADIUS, whatIsGround)) {
+			if (Physics2D.OverlapCircle (ceilingCheck.position, CEILING_RADIUS, whatIsGround))
+			{
 				crouch = true;
 			}
 		}
 
 		//only control the player if grounded or airControl is turned on
-		if (grounded || airControl) {
+		if (grounded || airControl)
+		{
 
 			// If crouching
-			if (crouch) {
+			if (crouch)
+			{
 				// Reduce the speed by the crouchSpeed multiplier
 				movement *= crouchSpeed;
 
 				// Disable one of the colliders when crouching
-				if (crouchDisableCollider != null) {
+				if (crouchDisableCollider != null)
+				{
 					crouchDisableCollider.enabled = false;
 				}
-			} else {
+			}
+			else
+			{
 				// Enable the collider when not crouching
-				if (crouchDisableCollider != null) {
+				if (crouchDisableCollider != null)
+				{
 					crouchDisableCollider.enabled = true;
 				}
 			}
 
 			// Move the character by finding the target velocity
-			Vector3 targetVelocity = new Vector2(movement * 10f, playerRigidbody2D.velocity.y);
+			Vector3 targetVelocity = new Vector2 (movement * 10f, playerRigidbody2D.velocity.y);
 			// And then smoothing it out and applying it to the character
-			playerRigidbody2D.velocity = Vector3.SmoothDamp(playerRigidbody2D.velocity, targetVelocity, ref velocity, movementSmoothing);
+			playerRigidbody2D.velocity = Vector3.SmoothDamp (playerRigidbody2D.velocity, targetVelocity, ref velocity, movementSmoothing);
 
 			// If the input is moving the player right and the player is facing left...
-			if (movement > 0 && !facingRight) {
+			if (!facingRight)
+			{
 				// ... flip the player.
-				Flip();
+				Flip ();
 			}
 			// Otherwise if the input is moving the player left and the player is facing right...
-			else if (movement < 0 && facingRight) {
+			else if (facingRight)
+			{
 				// ... flip the player.
-				Flip();
+				Flip ();
 			}
 		}
 		// If the player should jump...
-		if (grounded && jump) {
+		if (grounded && jump)
+		{
 			// Add a vertical force to the player.
 			grounded = false;
-			playerRigidbody2D.AddForce(new Vector2(0f, jumpForce));
+			playerRigidbody2D.AddForce (new Vector2 (0f, jumpForce));
 		}
 	}
 
-	private void Flip() {
-		// Switch the way the player is labelled as facing.
-		facingRight = !facingRight;
-
-		// Multiply the player's x local scale by -1.
-		Vector3 theScale = transform.localScale;
-		theScale.x *= -1;
-		transform.localScale = theScale;
+	private void Flip ()
+	{
+		// If the gun is facing to the right, flip character left
+		if (angle > -90 || angle < 90)
+		{
+			facingRight = false;
+			transform.rotation = Quaternion.Euler (new Vector3 (0, 180f, 0));
+		}
+		// If the gun is facing to the right, flip character right
+		if (angle < -90 || angle > 90)
+		{
+			facingRight = true;
+			transform.rotation = Quaternion.Euler (new Vector3 (0, 0, 0));
+		}
```

Comment:
Dette vil sette ny rotasjon på hver frame selvom rotasjonen er innen for samme område som ikke forårsaker flipping.
Finne en måte å gjøre en boolsk sjekk før flipp.